### ISSUE TITLE
Full ek55 test set results

### ISSUE THREAD

**Issue by r3ward** [debadityaroy/LatentGoal#2]
How do I implement your model on the full EK55 test set? I am aiming to gain the results shown in Table 3 in your paper. I have all the datasets and csv as you spoke about in the readme.
Thanks!

**Comment by debadityaroy** [comment]
Hi @r3ward 
You need to train two models - one for verb using TSNRGB features and another for noun using bag of object (OBJ) features obtained from RULSTM repository. 

- For verb model, change [`this line`](https://github.com/debadityaroy/LatentGoal/blob/2b4f7fcdce0bf1e46ea3ef5639bf28f32efe7919/tsnrgb_feat_latent_goal_action_max_current_action.py#L339
) to `self.embedding2verb` and for noun model to `self.embedding2noun`

**Comment by r3ward** [comment]
Thanks! May I have an email address to reach you on? Have a few more questions about the implimentation

**Comment by debadityaroy** [comment]
You can contact me on this address.

**Comment by r3ward** [comment]
- I trained both RGB and OBJ with the mentioned dimensions using tsnrgb_feat_latent_goal_action_max_current_action.py
- This put two separate .pt files into /ckp, I then placed all the parameters into latent_goal_inference_rgb+obj.py.

Just to mention a change i had to make: Lines 103 in latent_goal_inference_rgb+obj.py created a depreciated tensor error (only from the rgb feature) '.copy() - I switching it to '.clone()'.

When I run the program, I am getting a parser error when reading in 'seen.csv' and 'unseen.csv' into the evaluate function. The reading of the csv is throwing the error - 'expecting 1 fields in line 2, saw 2'. Any idea of how to circumvent this? I am attempting to get validation results with different anticipation times and the test results.

**Comment by debadityaroy** [comment]
1. I am not able to replicate your issue, I am able to parse both test_seen.csv and test_unseen.csv using the evaluate function. Are you using the csv files from [https://github.com/fpv-iplab/rulstm/tree/master/RULSTM/data/ek55](here)?

2. For different anticipation times, you need to add `ant_time = ant_sec*30` where `ant_sec` is the anticipation seconds (default 1.0s). Then change `stop = stop - ant_time` https://github.com/debadityaroy/LatentGoal/blob/cf49bc7b41f2f66cddcbdbca5dd3939e325cc084/latent_goal_inference_rgb%2Bobj.py#L89, commented code is already present.

**Comment by r3ward** [comment]
Thanks for the response, I'm retraining for obj and rgb in case I had an incorrect parameter somewhere - I'll let you know how I get on. Did you ever get results in Top-5 Action Accuracy for timesteps between 0.25 seconds and 2.0 seconds? I noticed a couple of models review this as part of their findings, so including it within my survey and would appreciate it if you have these figures. Additionally, how did you extract findings from the seen and unseen JSONs, I have a seen.json but trying to find a script to parse it for the final results. Thanks!

**Comment by debadityaroy** [comment]
The results on different anticipation times are reported on validation set, not test set.

**Comment by r3ward** [comment]
Right, I'm trying to get all the Top-5 Action accuracies ranging from 0.25 to 2.0 seconds on the validation set. Did you ever get these results? Collecting the results like this: [https://ibb.co/z62snVQ](url)

**Comment by debadityaroy** [comment]
I can't access your link.
